cmd/limactl/start.go

```diff
@@ -92,9 +94,16 @@ func loadOrCreateInstance(cmd *cobra.Command, args []string) (*store.Instance, e
 
 	if ok, u := argSeemsTemplateURL(arg); ok {
 		templateName := u.Host
+		if u.Path != "" {
+			templateName, err = securejoin.SecureJoin(templateName, u.Path)
+			if err != nil {
+				return nil, err
+			}
+		}
```

Comment:
It doesn't make sense to use `SecureJoin` here because it is still a relative path. And `filepath.Join` will just ignore empty elements, so no need for the conditional either:

```suggestion
		templateName = filepath.Join(u.Host, u.Path)
```

The real `SecureJoin` check happens later in `readTemplate` when this relative name is joined with the full path to the examples directory.

Reply:
IMHO it still makes sense to use `SecureJoin` 
https://go.dev/play/p/DYA2yxk4iBr


> u=url.Parse("template://foo/../bar"), u.Host="foo", u.Path="/../bar", filepath.Join(u.Host, u.Path)="bar", securejoin.SecureJoin(u.Host, u.Path="foo/bar")

Reply:
`SecureJoin` is kind of a light-weight `chroot` jail. It only makes sense when the left side of the "join" is internal to the app. When both parts are generated from user input, then secure join normally doesn't provide any additional protection.

So yes, `foo/../bar` would be treated as `bar`, but how is this a problem? The user could have written `template://bar` with the same result, so the file is already accessible.

The "chroot" for this use case is `/usr/local/share/lima/examples`, so any file under that root is accessible to the user, but since this root directory is built into the app itself, the user cannot escape it. This is enforced in `readTemplate`, because the root directory there is not based on user input.

Why should `foo/../bar` be `foo/bar`, but `foo/bar/../baz` is `foo/baz` and not `foo/bar/baz`? What makes the first directory level under `.../examples/` special, so it needs different treatment? It is breaking path semantics for no actual security benefit.

Reply:
fixed